### PR TITLE
Convert SymHop expressions to LaTeX

### PR DESCRIPTION
**Pull request by robbr48**
Can be tested by using the `latex` function in HCOM:
```
>> f=expr((a+b)/(c+d))
(a+b)/(c+d)
>> latex(f)
\dfrac{a+b}{c+d}
```

### CODE REVIEW

File: HopsanGUI/HcomHandler.cpp [hunk 1]
```diff
@@ -7261,7 +7261,7 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
             isHcomFunctionCall("removeDivisors", expr) || isHcomFunctionCall("left", expr) ||
             isHcomFunctionCall("right", expr) || isHcomFunctionCall("trapezoid", expr) ||
             isHcomFunctionCall("euler", expr) || isHcomFunctionCall("bdf1", expr) ||
-            isHcomFunctionCall("bdf2", expr)) {
+            isHcomFunctionCall("bdf2", expr) || isHcomFunctionCall("latex", expr)) {
```
Comment (peterNordin) [inline HopsanGUI/HcomHandler.cpp:7264]: Are any of these symhop/hcom functions part of the hcom documentation?

File: SymHop/src/SymHop.cpp [hunk 2]
```diff
@@ -1279,6 +1279,203 @@ QString Expression::toString() const
     return ret;
 }
 
+QString Expression::toLaTeX() const
+{
+    QString ret;
+
+    if(this->isSymbol())
+    {
+        QStringList greekSymbols = {"alpha","beta","gamma","Gamma","delta","Delta","epsilon","varepsilon","zeta","eta","theta","vartheta","Theta","iota","kappa","lambda","Lambda","mu","nu","xi","Xi","pi","Pi","rho","varrho","sigma","Sigma","tau","upsilon","Upsilon","phi","varphi","Phi","chi","psi","Psi","omega","Omega"};
```
Comment (peterNordin) [inline SymHop/src/SymHop.cpp:1288]: Some of these have only lower case version, maybe intended, I dont know

File: SymHop/src/SymHop.cpp [hunk 3]
```diff
@@ -1279,6 +1279,203 @@ QString Expression::toString() const
...
+            ret = QString::number(mString.toDouble());
+        }
+        else {
+            ret = mString;
+        }
+    }
+    else if(this->isFunction())
+    {
+        QStringList builtinFunctions = {"sin","cos","tan","arcsin","arccos","arctan","cot","sec","csc","sinh","cosh","tanh","lg","ln","log","min","max"     };
+        if(mFunction == "pi" && mArguments.size() == 0) {
```
Comment (peterNordin) [inline SymHop/src/SymHop.cpp:1303]: Is pi a function ?

File: SymHop/src/SymHop.cpp [hunk 4]
```diff
@@ -1279,6 +1279,203 @@ QString Expression::toString() const
...
+        {
+            powerStr.prepend("(");
+            powerStr.append(")");
+        }
+        if(powerStr.size() > 1) {
+            powerStr.prepend("{");
+            powerStr.append("}");
+        }
+        ret = baseStr+"^"+powerStr;
+        //ret = "pow("+baseStr+","+powerStr+")";
```
Comment (peterNordin) [inline SymHop/src/SymHop.cpp:1469]: comment

**Review by peterNordin — APPROVED** [review]
I just browsed through the code, look good.
I guess this is helpful when writing the markdown documentation ?
Pasting the c++ equations into hcom and converting to latex ?

File: SymHop/src/SymHop.cpp (hunk 2 above)
Comment (robbr48) [reply SymHop/src/SymHop.cpp:1288]: Yes, it is intentional. Many upper case greek letters are equivalent to latin letters, and those don't have latex commands.

File: SymHop/src/SymHop.cpp (hunk 3 above)
Comment (robbr48) [reply SymHop/src/SymHop.cpp:1303]: Yes, in SymHop, but not in Latex. Therefore `pi()` must be converted to `\pi`.

File: HopsanGUI/HcomHandler.cpp (hunk 1 above)
Comment (robbr48) [reply HopsanGUI/HcomHandler.cpp:7264]: No, you are right. They should be registered as internal functions. I will fix that.

**Pull request merged**